### ISSUE TITLE
`gr.Audio()` does not maintain value after being set to invisible in Gradio 3.2

### ISSUE THREAD

**Issue by oconnoob** [gradio-app/gradio#2156]
### Describe the bug

When working with Gradio 3.1.6, I would switch between file and microphone audio sources, setting the deselected component to be invisible. E.g.:

Step 1 - select "Record Audio":
![image](https://user-images.githubusercontent.com/46252299/187986321-5400ef0f-d796-42c4-b18c-a1fc413663bc.png)

Step 2 - record Audio:
![image](https://user-images.githubusercontent.com/46252299/187986813-18c4dd82-06a8-4bfd-b3ee-9b8bae70be1a.png)

Step 3 - switch to "Audio File"
- This sets the `mic_data` component (`gr.Audio()`) to invisible, and the previously invisible `file_data` component (`gr.Audio()`) to be visible. 
![image](https://user-images.githubusercontent.com/46252299/187986843-17054550-c77c-4e85-b081-d04ac752642a.png)

Step 4 - switch back to "Record Audio":
- This does the reverse of the Step 3, switching visibility as appropriate
**- The recorded audio file is still present as the value of the `mic_data` component**
![image](https://user-images.githubusercontent.com/46252299/187986878-fceacd33-1267-47f1-b1d1-78ac8c94f143.png)

Now, if I upgrade to `gradio==3.2` from `gradio==3.1.6`, **Step 4 now has a different result**. Namely, the audio file does not persist, and this is the result of switching back to "Record Audio" in Step 4 **without deleting the previously recorded audio.**

![image](https://user-images.githubusercontent.com/46252299/187987313-38d8e9b8-88e4-4fe1-af1c-635d74ea4eed.png)

Note that `gr.State()` was not working in 3.1.6, so I was temporarily using global variables, although think that I've done everything right updating to `gr.State()`s and therefore the problem is not stemming from any mistake there.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Simply running the above steps in the GUI will reproduce the result, selecting either the broken or working commit to see the corresponding behavior:

### BROKEN (Gradio 3.2):
Commit: 90b6c88d33e4efcd43aaef3a17eee71fc7dc328f

### WORKING (Gradio 3.1.6):
Commit: cf9607fbbe0af33b0bfc8e1a51a250c08aff123f

### Reproduce:
```bash
git clone https://github.com/AssemblyAI-Examples/audio-intelligence-dashboard.git
cd audio-intelligence-dashboard
git checkout <COMMIT>
cd audio-intelligence-dashboard
python -m virtualenv venv
.\venv\Scripts\activate.bat # (or MacOS equilvaent)
pip install -r requirements.txt
python app/app.py
```

### Logs

```shell
N/A
```

### System Info

```shell
Win11, Gradio 3.2, Edge/Chrome
```

### Severity

annoying

**Comment by freddyaboulton** [comment]
Hi @oconnoob ! Sorry about that. Seems like a duplicate of #2154 . Can you try the workaround in that issue in the meantime?

**Comment by oconnoob** [comment]
@freddyaboulton Oh you're right, my mistake. I was searching specifically for `gr.Audio()`!

Yes I can work around it in the meantime, thanks!

**Comment by abidlabs** [comment]
Thanks @oconnoob will close the issue for now then :)

**Comment by freddyaboulton** [comment]
Fixed in #2157
